src/control/controlproxy.cpp

```diff
@@ -3,35 +3,41 @@
 #include "control/controlproxy.h"
 #include "control/control.h"
 
-ControlProxy::ControlProxy(QObject* pParent)
-        : QObject(pParent),
-          m_pControl(NULL) {
+ControlProxy::ControlProxy(const QString& g, const QString& i, QObject* pParent, ControlFlags flags)
+        : ControlProxy(ConfigKey(g, i), pParent, flags) {
 }
 
-ControlProxy::ControlProxy(const QString& g, const QString& i, QObject* pParent)
-        : QObject(pParent) {
-    initialize(ConfigKey(g, i));
+ControlProxy::ControlProxy(const char* g, const char* i, QObject* pParent, ControlFlags flags)
+        : ControlProxy(ConfigKey(g, i), pParent, flags) {
 }
 
-ControlProxy::ControlProxy(const char* g, const char* i, QObject* pParent)
-        : QObject(pParent) {
-    initialize(ConfigKey(g, i));
-}
+ControlProxy::ControlProxy(const ConfigKey& key, QObject* pParent, ControlFlags flags)
+        : QObject(pParent),
+          m_pControl(nullptr) {
+    DEBUG_ASSERT(!key.isNull() || flags.testFlag(ControlFlag::AllowEmptyKey));
+    m_key = key;
 
-ControlProxy::ControlProxy(const ConfigKey& key, QObject* pParent)
-        : QObject(pParent) {
-    initialize(key);
+    if (!flags.testFlag(ControlFlag::InitializeLater)) {
+        flags.setFlag(ControlFlag::InitializeLater, false);
+        initialize(flags);
+    }
 }
 
-void ControlProxy::initialize(const ConfigKey& key, bool warn) {
-    m_key = key;
-    // Don't bother looking up the control if key is NULL. Prevents log spew.
-    if (!key.isNull()) {
-        m_pControl = ControlDoublePrivate::getControl(key, warn);
+void ControlProxy::initialize(ControlFlags flags) {
+    // Prevent double initialization
+    DEBUG_ASSERT(!m_pControl);
+
+    // Prevent empty keys
+    if (m_key.isNull()) {
+        DEBUG_ASSERT(flags.testFlag(ControlFlag::AllowEmptyKey));
+        return;
     }
+
+    m_pControl = ControlDoublePrivate::getControl(m_key, flags);
+    DEBUG_ASSERT(flags.testFlag(ControlFlag::NoAssertIfMissing) || m_pControl);
```

Comment:
We should decide for one order. First the condition followed the the flag test or the other way round? I would prefer to use `|| testFlag()` as a suffix, because the condition is what actually matters.